synapse/lib/cell.py

```diff
@@ -1177,6 +1177,14 @@ async def __anit__(self, dirn, conf=None, readonly=False, parent=None):
         self.cellinfo = await node.dict()
         self.onfini(node)
 
+        lastver = self.cellinfo.get('cell:version')
+        if lastver is None:
+            await self.cellinfo.set('cell:version', self.VERSION)
+
+        if lastver is not None and self.VERSION < lastver:
+            mesg = f'Version mismatch for {self.getCellType()}. Current version ({self.VERSION}) is less than previous version ({lastver}).'
+            raise s_exc.BadVersion(mesg=mesg, currver=self.VERSION, lastver=lastver)
+
```

Comment:
Need to bump the cellversion if necessary after checking that we didn't go backwards.